### PR TITLE
Fix derived dim bugs in ep.run_decomp

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123326

Differential Revision: [D55730289](https://our.internmc.facebook.com/intern/diff/D55730289)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55730289).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55730289).

**Review by avikchaudhuri — APPROVED** [review]

File: torch/export/exported_program.py [hunk 1]
```diff
@@ -700,7 +713,11 @@ def _get_updated_graph_signature(
                 self.graph_signature, transformed_gm
             ),
             state_dict=self.state_dict,
-            range_constraints=_get_updated_range_constraints(transformed_gm),
+            range_constraints=_get_updated_range_constraints(
+                transformed_gm,
+                self._num_lifted_params_buffers(),
```
Comment (tugsbayasgalan) [inline torch/export/exported_program.py:718]: cc: @gmagogsfm
